generator/src/main/java/com/scottlogic/deg/generator/generation/RegexStringGenerator.java

```diff
@@ -165,9 +163,51 @@ private String getMatchedString(int indexOrder) {
         }
         String result = buildStringFromNode(rootNode, indexOrder);
         result = result.substring(1, result.length() - 1);
+
+        /**
+         * <p>
+         *     FIXME - This check will be removed if/when the dk.brics.automaton
+         *     library is fixed to support surrogate pairs,
+         * </p>
+         * <p>
+         *     issue #15 (https://github.com/cs-au-dk/dk.brics.automaton/issues/15)
+         *     has been raised on the dk.brics.automaton library
+         * </p>
+         * <p>
+         *     issue #537 has been created to track when the dk.brics.automaton library
+         *     is updated.
+         * </p>
+         */
+        if (!containsValidUtf8Chars(result)) {
+            return null;
+        }
+
         return result;
     }
 
+    /**
+     * <p>
+     * check to see if the character generated is a valid utf-8 single word value.
+     * </p>
+     * <p>
+     * from chapter 3.9, page 126 of `the Unicode Standard v11.0`
+     * (https://www.unicode.org/versions/Unicode11.0.0/ch02.pdf):
+     * </p>
+     * <code>Because surrogate code points are not Unicode scalar values, any UTF-8 byte
+     * sequence that would otherwise map to code points U+D800..U+DFFF is illformed.
+     * </code>
+     */
+    public static boolean containsValidUtf8Chars(String str) {
+        if (str != null) {
```

Comment:
If you invert this `if` you can reduce the nesting within this function, which might make it a bit easier to follow.